dipy/workflows/segment.py

```diff
@@ -368,3 +372,144 @@ def run(
+        b0_threshold : float, optional
+            The intensity threshold for a b=0 image. used only for 'dam' method.
+        low_signal_threshold : float, optional
+            The threshold below which a voxel is considered to have low signal.
+            Used only for 'dam' method.
+        nclasses : int, optional
+            Number of desired classes. Used only for 'hmrf' method.
+        beta : float, optional
+            Smoothing parameter, the higher this number the smoother the
+            output will be. Used only for 'hmrf' method.
+        tolerance : float, optional
+            Value that defines the percentage of change tolerated to
+            prevent the ICM loop to stop. Default is 1e-05.
+            If you want tolerance check to be disabled put 'tolerance = 0'.
+            Used only for 'hmrf' method.
+        max_iter : int, optional
+            Fixed number of desired iterations. Default is 100.
+            This parameter defines the maximum number of iterations the
+            algorithm will perform. The loop may terminate early if the
+            change in energy sum between iterations falls below the
+            threshold defined by `tolerance`. However, if `tolerance` is
+            explicitly set to 0, this early stopping mechanism is disabled,
+            and the algorithm will run for the specified number of
+            iterations unless another stopping criterion is met.
+            Used only for 'hmrf' method.
+        out_dir : string, optional
+            Output directory.
+        out_tissue : string, optional
+            Name of the tissue volume to be saved.
+        out_pve : string, optional
+            Name of the pve volume to be saved.
+
+        REFERENCES
+        ----------
+        .. footbibliography::
+
+        """
+        io_it = self.get_io_iterator()
+
+        if not method or method.lower() not in ["hmrf", "dam"]:
+            logging.error(
+                f"Unknown method '{method}' for tissue extraction. "
+                "Choose '--method hmrf' (for T1w) or '--method dam' (for DWI)"
+            )
+            sys.exit(1)
+
+        prefix = "t1" if method.lower() == "hmrf" else "dwi"
+        for i, name in enumerate(self.flat_outputs):
+            if name.endswith("tissue_classified.nii.gz"):
+                self.flat_outputs[i] = name.replace(
+                    "tissue_classified.nii.gz", f"{prefix}_tissue_classified.nii.gz"
+                )
+            if name.endswith("tissue_classified_pve.nii.gz"):
+                self.flat_outputs[i] = name.replace(
+                    "tissue_classified_pve.nii.gz",
+                    f"{prefix}_tissue_classified_pve.nii.gz",
+                )
+
+        self.update_flat_outputs(self.flat_outputs, io_it)
+
+        for fpath, tissue_out_path, opve in io_it:
+            logging.info(f"Extracting tissue from {fpath}")
+
+            data, affine, img = load_nifti(fpath, return_img=True)
```

Comment:
We are loading the img here but we do not use it

Reply:
Thanks, fixed!